plugins/utils/todo-comments.nix

```diff
@@ -16,275 +24,394 @@ let
+              description = "Specify the directory to search for comments";
+              default = null;
+              example = "~/projects/foobar";
+            };
+
+            keywords = mkOption {
+              type = types.nullOr types.str;
+              description = ''
+                Comma separated list of keywords to filter results by.
+                Keywords are case-sensitive.
+              '';
+              default = null;
+              example = "TODO,FIX";
+            };
+
+            options = keymaps.mapConfigOptions;
+          };
+      in
+      mapAttrs mkKeymapOption commands;
+
+    ripgrepPackage = mkPackageOption {
+      name = "ripgrep";
+      default = pkgs.ripgrep;
     };
   };
 
-  config =
-    let
-      setupOptions = {
-        inherit (cfg) signs;
-        sign_priority = cfg.signPriority;
-        inherit (cfg) keywords;
-        gui_style = cfg.guiStyle;
-        merge_keywords = cfg.mergeKeywords;
-        highlight = {
-          inherit (cfg.highlight)
-            multiline
-            before
-            keyword
-            after
-            exclude
-            ;
-          pattern = helpers.ifNonNull' cfg.highlight.pattern (helpers.mkRaw "[[${cfg.highlight.pattern}]]");
-          multiline_pattern = cfg.highlight.multilinePattern;
-          multiline_context = cfg.highlight.multilineContext;
-          comments_only = cfg.highlight.commentsOnly;
-          max_line_len = cfg.highlight.maxLineLen;
-        };
-        inherit (cfg) colors;
-        search = {
-          inherit (cfg.search) command args;
-          pattern = helpers.ifNonNull' cfg.search.pattern (
-            if isList cfg.search.pattern then
-              (map helpers.mkRaw cfg.search.pattern)
-            else
-              helpers.mkRaw "[[${cfg.search.pattern}]]"
-          );
-        };
-      } // cfg.extraOptions;
-    in
-    mkIf cfg.enable {
-      extraPlugins = [ cfg.package ];
-      extraPackages = [ cfg.ripgrepPackage ];
-      extraConfigLua = ''
-        require("todo-comments").setup${helpers.toLuaObject setupOptions}
-      '';
+  extraConfig = cfg: {
+    extraPackages = [ cfg.ripgrepPackage ];
+
+    keymaps = flatten (
+      mapAttrsToList (
+        optionName: funcName:
+        let
+          keymap = cfg.keymaps.${optionName};
+
+          cwd = optionalString (keymap.cwd != null) " cwd=${keymap.cwd}";
+          keywords = optionalString (keymap.keywords != null) " keywords=${keymap.keywords}";
+        in
+        optional (keymap != null) {
+          mode = "n";
+          inherit (keymap) key options;
+          action = ":${funcName}${cwd}${keywords}<CR>";
+        }
+      ) commands
+    );
+
+    # Automatically enable plugins if keymaps have been set
+    plugins = mkMerge [
+      (mkIf (cfg.keymaps.todoTrouble != null) { trouble.enable = true; })
+      (mkIf (cfg.keymaps.todoTelescope != null) { telescope.enable = true; })
+    ];
```

Comment:
Is this something we still want to do, or would we rather warn/assert asking the user to enable the plugins themselves?

Reply:
This is something I saw and mentioned in https://github.com/nix-community/nixvim/pull/1559. I wasn't sure what everyone's opinions were so I left it untouched. But, based on that PR it would seem we'd want to change it to assertions.